### PR TITLE
Add checkstyle

### PR DESCRIPTION
**Pull request by Stypox**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

This PR adds checkstyle to the project. The motivation for doing this, and doing it in a single PR is the following: there have been many PRs lately that should fix one thing but while they are at it they also fix the formatting, making it far more difficult to review. That's why from now on any misformatting will be reported as an **error** (only `test` tasks though, not `compileJava` tasks so that we don't overload Jitpack's servers with useless checkstyle checks).

- In commits "...downloader.Request", "...DashMpdParser", "...DonationLinkHelper", "...JsonUtils" I refactored some other small things (explained in the commits' descriptions). I feel bad for this as they are somewhat unrelated changes, but it would probably have taken me more time to reformat than to refactor ;-)
- In commits "...MediaFormat..." and "...NewPipe..." instead of reformatting some methods I converted them to Java 8 Streams API altogether for the same reason as above ;-)
- The `checkstyle.xml` I used is the same as NewPipe minus some Javadoc-related stuff. There is no need for strict rules over comments here in the extractor, as sometimes javadocs are just needed to clarify a small thing and having empty/meaningless @param or @throws is useless.
- I had to use checkstyle version 9.3 instead of 10.0 as the latest requires compile SDK 11
- While doing this PR I also found other places where the streams API could be used, but I refrained from adding unrelated changes here and will open a PR after this is merged. *Please remind me if I forget when merging (and a reminder for myself: the local branch is called `fix-warnings`).*

After this PR is merged I would propose to **BAN any reformatting** unrelated to the modified code in extractor PRs, unless the PR contains *only* reformats. The ways to format things that are not constrained by checkstyle are pretty subjective (e.g. whether to put multiple `.foo().bar().baz()` on the same line or split them), so it does not make sense to keep reformatting them, and in any case it should be done in separate PRs.

I hope this PR will be merged soon otherwise... *many* conflicts to solve ;-)

### CODE REVIEW

**Comment by opusforlife2** [comment]
> I hope this PR will be merged soon

Ahahahahahahahahahahaha

File: extractor/src/main/java/org/schabi/newpipe/extractor/utils/DonationLinkHelper.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package org.schabi.newpipe.extractor.utils;
 
 import java.net.MalformedURLException;
-import java.net.URL;
 
-public class DonationLinkHelper {
+/**
+ * Note: this class seems unused? Should it be removed?
+ */
+public final class DonationLinkHelper {
```
Comment (litetex) [inline extractor/src/main/java/org/schabi/newpipe/extractor/utils/DonationLinkHelper.java:8]: Yes.
Please check if it causes any problems in NewPipe

**Review by litetex — COMMENTED** [review]
Good work! ``(；^＿^)ッ☆(　゜u゜)``

Took me some time to review this... but mostly LGTM, only some minor things:

PS: Also please rebase the PR

File: extractor/src/main/java/org/schabi/newpipe/extractor/utils/DonationLinkHelper.java (hunk 1 above)
Comment (litetex) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/DonationLinkHelper.java:8]: Checked: Can be removed without problems

File: extractor/src/main/java/org/schabi/newpipe/extractor/linkhandler/ListLinkHandlerFactory.java [hunk 2]
```diff
@@ -13,17 +13,23 @@
...
 
-    public List<String> getContentFilter(String url) throws ParsingException {
+    @SuppressWarnings({"RedundantThrows", "unused"})
+    public List<String> getContentFilter(final String url) throws ParsingException {
         return Collections.emptyList();
     }
 
-    public String getSortFilter(String url) throws ParsingException {
+    @SuppressWarnings({"RedundantThrows", "unused"})
+    public String getSortFilter(final String url) throws ParsingException {
```
[the remark this answers is not in the document]
Comment (litetex) [reply extractor/src/main/java/org/schabi/newpipe/extractor/linkhandler/ListLinkHandlerFactory.java:22]: Checked: Methods are unused and can be removed.

``contentFilters`` and ``sortfilter`` are get inside the ``ListLinkHandler`` and not the ``ListLinkHandlerFactory``
``ListLinkHandlerFactory`` only passes these values through when ``fromQuery`` is called

**Comment by litetex** [comment]
Fixed above problems myself.

**Comment by litetex** [comment]
All Mix-Playlist tests are failing:
```
Could not get playlistData
org.schabi.newpipe.extractor.exceptions.ExtractionException: Could not get playlistData
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeMixPlaylistExtractor.onFetchPage(YoutubeMixPlaylistExtractor.java:100)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:60)
	at org.schabi.newpipe.extractor.services.youtube.YoutubeMixPlaylistExtractorTest$ChannelMix.setUp(YoutubeMixPlaylistExtractorTest.java:356)
	...
```

**Comment by TobiGr** [comment]
Do you want to merge this before #810? I would not suggest it.

**Comment by litetex** [comment]
@TobiGr 
Yes

**Comment by litetex** [comment]
Current test-results (mock):

grafik

Reasons:
* HlsOnlyStreams → Peertube instance (https://tilvids.com) is down (Status code: 502)
* see #813 for the rest

**Review by litetex — APPROVED** [review]

**Pull request merged**